pkg/gate/gate.go

```diff
@@ -12,67 +12,138 @@ import (
 	promgate "github.com/prometheus/prometheus/pkg/gate"
 )
 
-// Gate is an interface that mimics prometheus/pkg/gate behavior.
+var (
```

Comment:
Although I get the reasoning behind this change, the whole idea behind introducing the keeper was to remove metrics as package level variables. I wonder could we introduce some other pattern to avoid this.

Reply:
these are only variables for metric description. The metrics themselves are created in `InstrumentGateDuration()` and `InstrumentGateInFlight`.

Reply:
Oh yes, sorry 🤦‍♂️